tstest/clock.go

```diff
@@ -4,57 +4,495 @@
 package tstest
 
 import (
+	"container/heap"
 	"sync"
 	"time"
+
+	"tailscale.com/tstime"
+	"tailscale.com/util/mak"
 )
 
+// ClockBuilder is used to configure the initial settings for a Clock. Once
+// the settings are configured as desired, call ClockBuilder.Finish to get the
+// resulting Clock. ClockBuilder can only be used to create a single Clock.
+type ClockBuilder struct {
+	c *Clock
+}
+
+func NewClockBuilder() *ClockBuilder {
+	cb := &ClockBuilder{
+		c: new(Clock),
+	}
+	return cb
+}
+
+func (cb *ClockBuilder) Start(t time.Time) *ClockBuilder {
+	cb.c.start = t
+	return cb
+}
+
+func (cb *ClockBuilder) Step(d time.Duration) *ClockBuilder {
+	cb.c.step = d
+	return cb
+}
+
+func (cb *ClockBuilder) TimerChannelSize(n int) *ClockBuilder {
+	cb.c.timerChannelSize = n
+	return cb
+}
+
+func (cb *ClockBuilder) Finish() *Clock {
+	c := cb.c
+	cb.c = nil
+	c.init()
+	return c
+}
+
+// NewClock returns a Clock initialized with default settings (equivalent to
+// NewClockBuilder().Finish()).
```

Comment:
Make the implementation literally NewClockBuilder().Finish() ? That way if we ever change defaults we don't need to remember to update two places